File: 392.is-subsequence.cs
// @leet start
public class Solution {
    public bool IsSubsequence(string s, string t) {
      if(s.Length == 0) return true;
      var j = 0;
      for(int i = 0; i < t.Length; i++) {
        if(s[j] == t[i]) j++;
        if(j == s.Length) return true;
      }
      return false;
    }
}
// @leet end

File: 392.is-subsequence.kt
// @leet start
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
      if (s.isEmpty()) return true
      var j = 0
      for (i in t.indices) {
        if (s[j] == t[i]) j++
        if (j == s.length) return true
      }
      return false
    }
}
// @leet end
